src/task/methods/ambros_m_judges_third_place/script.py

```diff
@@ -0,0 +1,527 @@
+# This script is based on an IPython notebook:
+# https://github.com/Ambros-M/Single-Cell-Perturbations-2023/blob/main/notebooks/scp-26-py-boost-recommender-system-and-et.ipynb
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+from colorama import Fore, Back, Style
+from numpy.polynomial import Polynomial
+from scipy.stats import norm, skew, kurtosis
+import scipy
+from sklearn.linear_model import Ridge
+from sklearn.decomposition import PCA
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.neighbors import KNeighborsRegressor
+from sklearn.ensemble import ExtraTreesRegressor
+from itertools import combinations
+import cupy as cp
+from py_boost import GradientBoosting
+import warnings
+import zipfile
+
+warnings.simplefilter('ignore', FutureWarning)
+np.set_printoptions(linewidth=195, edgeitems=3)
+pd.set_option("min_rows", 6)
+
+## Viash start
+
+par = dict(
+    de_train_path = "/home/jovyan/git-repos/task-dge-perturbation-prediction/resources/neurips-2023-kaggle/de_train.parquet",
+    train_obs_zip_path = "/home/jovyan/git-repos/task-dge-perturbation-prediction/resources/neurips-2023-kaggle/train_obs.csv.zip",
+    id_map_path = "/home/jovyan/git-repos/task-dge-perturbation-prediction/resources/neurips-2023-kaggle/id_map.csv",
+    output_path = "output.parquet",
+)
+
+
+## Utility functions
+
+def mean_rowwise_rmse(y_true, y_pred):
+    """Competition metric
+    
+    Calling convention like in sklearn.metrics
+    """
+    mrrmse = np.sqrt(np.square(y_true - y_pred).mean(axis=1)).mean()
+    return mrrmse
+
+def de_to_t_score(de):
+    """Convert log10pvalues to t-scores
+    
+    Parameter:
+    de: array or DataFrame of log10pvalues
+    
+    Return value:
+    t_score: array or DataFrame of t-scores
+    """
+    p_value = 10 ** (-np.abs(de))
+#     return - scipy.stats.t.ppf(p_value / 2, df=420) * np.sign(de)
+    return - norm.ppf(p_value / 2) * np.sign(de)
+
+def t_score_to_de(t_score):
+    """Convert t-scores to log10pvalues (inverse of de_to_t_score)
+    
+    Parameter:
+    t_score: array or DataFrame of t-scores
+    
+    Return value:
+    de: array or DataFrame of log10pvalues
+    """
+#     p_value = scipy.stats.t.cdf(- np.abs(t_score), df=420) * 2
+    p_value = norm.cdf(- np.abs(t_score)) * 2
+    p_value = p_value.clip(1e-180, None)
+    return - np.log10(p_value) * np.sign(t_score)
+
+## Loading data
+
+de_train = pd.read_parquet(par['de_train_path'])
+id_map = pd.read_csv(par['id_map_path'], index_col = 0)
```

Comment:
Rename args in line with what is defined in `../api/comp_method.yaml`.

```suggestion
de_train_h5ad = pd.read_parquet(par['de_train_h5ad'])
id_map = pd.read_csv(par['id_map'], index_col = 0)
```

Reply:
In this case I should use `de_train = pd.read_parquet(par['de_train'])`, right? The method uses the parquet version of `de_train`, not the .h5ad version